docs/step-by-step/kubeapps-on-tkg/README.md

```diff
@@ -0,0 +1,72 @@
+
+As this guide will cover the installation of Kubeapps in a [VMware Tanzu™ Kubernetes Grid™](https://tanzu.vmware.com/kubernetes-grid) cluster and the configuration of an application repository for [VMware Tanzu™ Application Catalog™ (TAC)](https://tanzu.vmware.com/application-catalog), it assumes that you have access to a TKG cluster as well as a TAC pre-built or custom catalog. If you do not have access, please reach out to your [VMware sales representative](https://www.vmware.com/company/contact_sales.html).
+
+This tutorial, therefore, assumes that:
+
+- You have a Tanzu™ Kubernetes Grid v1.3.1 or later cluster properly configured with identy management. Check out the [VMware Tanzu™ Kubernetes Grid 1.3 Documentation](https://docs.vmware.com/en/VMware-Tanzu-Kubernetes-Grid/1.3/vmware-tanzu-kubernetes-grid-13/GUID-index.html) and the [Enabling Identity Management in Tanzu Kubernetes Grid](https://docs.vmware.com/en/VMware-Tanzu-Kubernetes-Grid/1.3/vmware-tanzu-kubernetes-grid-13/GUID-mgmt-clusters-enabling-id-mgmt.html) guide.
+- You have access to the [VMWare Cloud Services Portal (CSP)](https://console.cloud.vmware.com/). If not, talk to your [VMware sales representative](https://www.vmware.com/company/contact_sales.html) to request access.
+- You have access to, at least, the Tanzu™ Application Catalog™ Demo environment. If not, reach out to your [VMware sales representative](https://www.vmware.com/company/contact_sales.html).
+- You have the _kubectl_ CLI and the Helm v3.x package manager installed. Learn how to [install kubectl and Helm v3.x](https://docs.bitnami.com/kubernetes/get-started-kubernetes/#step-3-install-kubectl-command-line).
+
+## Intended Audience
+
+This information is intended for administrators who want to install Kubeapps on a Tanzu Kubernetes Grid cluster and use it to deploy and manage applications from the VMware Marketplace™ and the VMware Tanzu™ Application Catalog™. Besides, this content is also intended for application administrators and developers who want to use Kubeaps to deploy and manage modern apps in a Kubernetes architecture.
+
+The information is written for users who have a basic understanding of Kubernetes and are familiar with container deployment concepts. In-depth knowledge of Kubernetes is not required.
+
+# Steps
+
+<!-- To be extracted to a separate file -->
+
+## Step 1: Prepare the identity management
+
+<!-- Work in progress -->
+
+### Install or use the existing Pinniped in TKG?
+
+- [Enabling Identity Management in Tanzu Kubernetes Grid](https://docs.vmware.com/en/VMware-Tanzu-Kubernetes-Grid/1.3/vmware-tanzu-kubernetes-grid-13/GUID-mgmt-clusters-enabling-id-mgmt.html).
+- [Tanzu Kubernetes Grid 1.3 with Identity Management](https://liveandletlearn.net/post/kubeapps-on-tanzu-kubernetes-grid-13/)
+
+### Configure an OIDC provider
+
+- [VMware Cloud Services as OIDC provider](https://github.com/kubeapps/kubeapps/blob/master/docs/user/using-an-OIDC-provider.md#vmware-cloud-services)
```

Comment:
I think we need to be very careful saying this is a production configuration, given that using VMware Cloud Services Portal as an OIDC provider requires using an insecure tls setting. If it is ultra important to use VMware CSP, I think we should fix the insecure tls issue potentially by writing a custom provider, based on the oidc one, for the oauth2-proxy project that understands the different hierarchical issuers (it may also help us sort out the other issue we have with VMware CSP as the oidc provider of getting useful groups in the token).

Related to this, if the person following this has already completed the steps to setup identity management in TKG, following the VMware tutorial, then they'll already have setup Okta as the identity provider. Of course, that doesn't stop us including instructions for adding CSP as the identity provider, but I do wonder why there are no TKG docs that we've found for doing so.

Reply:
(oops, I forgot to click on comment and shut down my laptop :disappointed: , writing it again)

You're right, we can safely change it with "ready-to-use configuration" instead of using "production". However, the TLS thing is quite important, it's a pity CSP is not a fully OIDC-compliant provider. 
I'd personally rather avoid writing our custom provider in oauth2proxy,  Perhaps it's too much from the Kubeapps' point of view.

The good point about TKG  v1.3.1 and later is that Pinniped no longer requires Dex and uses the Pinniped endpoint for configuring OIDC provider. In the docs, they mention Okta, but just as an example. I think we should mention how to configure CSP (even if it requires the _insecure_ flag). Later on, we can either refer people to the generic docs or create more step-by-step docs with detailed information in each provider.

Anyway, my plan is, for now, to create the docs for CSP (disclaimer included) and confirm internally which OIDC provider we really want to use. In any case, CSP docs are also useful for us.